### PR TITLE
Followup on #2861

### PR DESCRIPTION
**Pull request by jedrazb**
## Follow up on unaddressed comments from #2861 

Addressed items:
- added debug logs in connector record manager
- moved `connector_exists` to ConnectorIndex class
- moved `generate_random_id` to utils class
- adapted tests

### CODE REVIEW

File: connectors/agent/connector_record_manager.py [hunk 1]
```diff
@@ -30,25 +28,27 @@ async def ensure_connector_records_exist(self, agent_config, connector_name=None
         """
 
         if not self._agent_config_ready(agent_config):
+            logger.debug(
+                "Agent configuration is not ready to create a connector record."
+            )
```
Comment (artem-shelkovnikov) [inline connectors/agent/connector_record_manager.py:33]: Can we update the copy to provide more specific detail on what's going on? How can configuration be not ready? Is it gonna be happening a lot?

File: connectors/protocol/connectors.py [hunk 2]
```diff
@@ -172,6 +173,18 @@ async def connector_put(
             index_name=index_name,
         )
 
+    async def connector_exists(self, connector_id):
+        try:
+            doc = await self.fetch_by_id(connector_id)
+            return doc is not None
```
Comment (artem-shelkovnikov) [inline connectors/protocol/connectors.py:179]: It's probably fine now, but is there a way to do this with as little traffic as possible? We can have connector records with cursors that'll weigh up to 5MB, personally I expect a method like "exists" to be veeeery lightweight

Comment (jedrazb) [reply connectors/protocol/connectors.py:179]: Agreed, it should be lightweight, there is not much we can do in this PR as we would need to adapt the API to allow to query only for selected fields.

I filed an issue: https://github.com/elastic/connectors/issues/2937

File: connectors/agent/connector_record_manager.py (hunk 1 above)
Comment (jedrazb) [reply connectors/agent/connector_record_manager.py:33]: > Is it gonna be happening a lot?

It should not be happening a lot, perhaps only once the service starts up with default values.  As soon as we receive first checkin the configuration should be "ready"  ... but this is extra check to ensure that the expected key/value pairs are actually there.

**Review by artem-shelkovnikov — APPROVED** [review]

File: connectors/agent/connector_record_manager.py [hunk 3]
```diff
@@ -62,43 +64,42 @@ async def ensure_connector_records_exist(self, agent_config, connector_name=None
...
         """
+
         connectors = agent_config.get("connectors")
-        if connectors is None or len(connectors) == 0:
-            return False
+        if connectors is None:
+            return False, "No 'connectors' key found in the service configuration."
+
+        if len(connectors) == 0:
+            return False, "Empty connectors array found in the service configuration."
```
Comment (artem-shelkovnikov) [inline connectors/agent/connector_record_manager.py:80]:
```suggestion
            return False, "Empty `connectors` array found in the service configuration."
```

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**